src/AggregateFunctions/AggregateFunctionDeepMergeJSON.cpp

```diff
@@ -0,0 +1,390 @@
+        validatePathLength(path_str.size());
+
+        size_t row_order;
+        readVarUInt(row_order, buf);
+
+        bool is_deleted;
+        readBinary(is_deleted, buf);
+
+        String value_str;
+        readStringBinary(value_str, buf);
+
+        total_size += path_str.size() + value_str.size();
+        validateTotalSize(total_size);
+
+        /// Deserialize Field
+        ReadBufferFromString value_buf(value_str);
+        Field value = readFieldBinary(value_buf);
+
+        auto interned_path = internString(path_str, arena);
+        aggregate_data.paths[interned_path] = DeepMergeJSONAggregateData::PathData{value, row_order, is_deleted};
+    }
+}
+
+void AggregateFunctionDeepMergeJSON::insertResultInto(AggregateDataPtr __restrict place, IColumn & to, [[maybe_unused]] Arena * arena) const
+{
+    const auto & aggregate_data = data(place);
+    Object result_object;
+
+    /// Helper function to set a value in a nested object structure
+    auto set_nested_value = [](Object & obj, const std::string & path, const Field & value)
+    {
+        size_t pos = 0;
+        Object * current = &obj;
+
+        while (pos < path.length())
+        {
+            size_t dot_pos = path.find('.', pos);
+            std::string key = (dot_pos == std::string::npos) ? path.substr(pos) : path.substr(pos, dot_pos - pos);
+
+            if (dot_pos == std::string::npos)
+            {
+                (*current)[key] = value;
+                break;
+            }
+
+            // Intermediate component - ensure it's an object
+            if (!current->contains(key) || current->at(key).getType() != Field::Types::Object)
+                (*current)[key] = Object();
+
+            Field & field = (*current)[key];
+            current = &field.safeGet<Object>();
+            pos = dot_pos + 1;
+        }
+    };
+
+    /// Convert flat paths back to nested object structure
+    for (const auto & [path_ref, path_data] : aggregate_data.paths)
+    {
+        if (!path_data.is_deleted && !aggregate_data.isObjectPath(path_ref))
+            set_nested_value(result_object, path_ref.toString(), path_data.value);
+    }
+
+    to.insert(result_object);
+}
+
+void AggregateFunctionDeepMergeJSON::addBatchSinglePlace(
+    size_t row_begin, size_t row_end, AggregateDataPtr __restrict place, const IColumn ** columns, Arena * arena, ssize_t if_argument_pos)
+    const
+{
+    if (if_argument_pos >= 0)
+    {
+        IAggregateFunctionDataHelper<DeepMergeJSONAggregateData, AggregateFunctionDeepMergeJSON>::addBatchSinglePlace(
+            row_begin, row_end, place, columns, arena, if_argument_pos);
+        return;
+    }
+
+    const auto & col_object = assert_cast<const ColumnObject &>(*columns[0]);
+    auto & aggregate_data = data(place);
+
+    for (size_t row = row_begin; row < row_end; ++row)
+    {
+        processColumnObject(col_object, row, aggregate_data, arena);
+        ++aggregate_data.row_count;
+    }
```

Comment:
We can do it more efficiently. For typed paths we can just add last value in the column. For dynamic paths we can add last non-null value. We will need to iterate only over shared data paths.

Reply:
Thanks for the optimization suggestion! I tried implementing it but discovered it breaks deletion markers that appear in intermediate rows. For example:

```sql
(1, '{"config": {"setting1": "value1"}}'),
(1, '{"config": {"setting1.$remove": true}}'),  -- This gets skipped
(1, '{"config": {"setting3": "value3"}}')
```

The optimization only processes the last row, missing the deletion marker in row 2. Would it be valuable to implement this optimization conditionally when no deletion key is configured? I think it will be, would it be ok if I did it in a follow-up PR?